### PR TITLE
Tag code samples for usage in developer docs

### PR DESCRIPTION
**Pull request by cbetta**
I've tagged the majority of code samples to an endpoint for usage in the new developer documentation.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @cbetta has signed the CLA. Thanks for the pull request!

**Comment by cbetta** [comment]
I'd be happy to merge this @mattwiller

File: docs/usage/files.md [hunk 1]
```diff
@@ -698,6 +715,7 @@ instance.
 Note: This method will only succeed if the provided metadata template is not currently applied to the file, otherwise it will 
 fail with a Conflict error.
 
+<!-- sample post_files_id_metadata_id_id -->
```
Comment (carycheng) [inline docs/usage/files.md]: maybe post_files_files_id_metadata_scope_template

File: docs/usage/files.md [hunk 2]
```diff
@@ -720,6 +738,7 @@ Note: This method will only succeed if the provided metadata template has alread
 have existing metadata, this method will fail with a Not Found error. This is useful you know the file will already have metadata applied,
 since it will save an API call compared to `set()`.
 
+<!-- sample put_files_id_metadata_id_id -->
```
Comment (carycheng) [inline docs/usage/files.md]: put_files_files_id_metadata_scope_template

File: docs/usage/files.md [hunk 3]
```diff
@@ -748,6 +767,7 @@ To retrieve the metadata instance on a file for a specific metadata template, fi
 metadata template to retrieve, then call [`metadata.get()`][metadata_get] to retrieve the metadata values attached to
 the file.  This method returns a `dict` containing the applied metadata instance.
 
+<!-- sample get_files_id_metadata_id_id -->
```
Comment (carycheng) [inline docs/usage/files.md]: get_files_files_id_metadata_scope_template

File: docs/usage/files.md [hunk 4]
```diff
@@ -763,6 +783,7 @@ To remove a metadata instance from a file, call
 metadata template to remove, then call [`metadata.delete()`][metadata_delete] to remove the metadata from the file.
 This method returns `True` to indicate that the removal succeeded.
 
+<!-- sample delete_files_id_metadata_id_id -->
```
Comment (carycheng) [inline docs/usage/files.md]: delete_files_files_id_metadata_scope_template

File: docs/usage/folders.md [hunk 5]
```diff
@@ -208,6 +214,7 @@ metadata instance.
 Note: This method will only succeed if the provided metadata template is not currently applied to the folder, otherwise it will 
 fail with a Conflict error.
 
+<!-- sample post_folders_id_metadata_id_id -->
```
Comment (carycheng) [inline docs/usage/folders.md]: post_folders_folders_id_metadata_scope_template

File: docs/usage/folders.md [hunk 6]
```diff
@@ -230,6 +237,7 @@ Note: This method will only succeed if the provided metadata template has alread
 have existing metadata, this method will fail with a Not Found error. This is useful you know the file will already have metadata applied,
 since it will save an API call compared to `set()`.
 
+<!-- sample put_folders_id_metadata_id_id -->
```
Comment (carycheng) [inline docs/usage/folders.md]: put_folders_folders_id_metadata_scope_template

File: docs/usage/folders.md [hunk 7]
```diff
@@ -258,6 +266,7 @@ To retrieve the metadata instance on a folder for a specific metadata template,
 the metadata template to retrieve, then call [`metadata.get()`][metadata_get] to retrieve the metadata values attached
 to the folder.  This method returns a `dict` containing the applied metadata instance.
 
+<!-- sample get_folders_id_metadata_id_id -->
```
Comment (carycheng) [inline docs/usage/folders.md]: get_folders_folders_id_metadata_scope_template

File: docs/usage/folders.md [hunk 8]
```diff
@@ -273,6 +282,7 @@ To remove a metadata instance from a folder, call
 metadata template to remove, then call [`metadata.delete()`][metadata_delete] to remove the metadata from the folder.
 This method returns `True` to indicate that the removal succeeded.
 
+<!-- sample delete_folders_id_metadata_id_id -->
```
Comment (carycheng) [inline docs/usage/folders.md]: delete_folders_folders_id_metadata_scope_template

**Comment by mattwiller** [comment]
@carycheng See [my comment in the Java SDK repo](https://github.com/box/box-java-sdk/pull/698#issuecomment-491456227) — the metadata comment tags should use `id` for now instead of `scope` and `template`.

**Pull request merged**